hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/LeaseManager.java

```diff
@@ -515,6 +508,13 @@ public void setLeasePeriod(long softLimit, long hardLimit) {
     this.softLimit = softLimit;
     this.hardLimit = hardLimit; 
   }
+
+  private synchronized Collection<Lease> getExpiredCandidateLeases() {
+    final long now = Time.monotonicNow();
+    return leases.values().stream()
+        .filter(lease -> lease.expiredHardLimit(now))
+        .collect(Collectors.toCollection(HashSet::new));
+  }
```

Comment:
I much prefer the loop in @daryn-sharp's original code.

    Collection<Lease> expired = new HashSet<>();
    for (Lease lease : leases) {
      if (lease.expiredHardLimit(now)) {
        expired.add(lease);
      }
    }
This streams code will have to change if we want to pull this back to branch 2.
I think @daryn-sharp  also said that stream()'s are more expensive.